syntax = "proto3";

service ProductImageAnalyzer {
  rpc GenerateFromImage (ImageRequest) returns (ImageResponse);
  rpc GenerateFromImageUrl (ImageUrlRequest) returns (ImageResponse);
  rpc HealthCheck (HealthCheckRequest) returns (HealthCheckResponse);
}

message ImageRequest {
  bytes image = 1;
  string filename = 2;
  string content_type = 3;
}

message ImageUrlRequest {
  string image_url = 1;
}

message ImageResponse {
  string title = 1;
  string description = 2;
}

message HealthCheckRequest {}

message HealthCheckResponse {
  string status = 1;
  string service = 2;
}